test/spec/provider.spec.js

```diff
@@ -146,4 +146,26 @@
             expect(b.container.Zero).toBe(0);
         });
     });
+    describe('Bottle#resetProviders', function() {
+        it('allows for already instantiated providers to be reset back to their registry', function() {
+            var i = 0;
+            var b = new Bottle();
+            var ThingProvider = function() { i = ++i; this.$get = function() { return this; }; };
+            b.provider('Thing', ThingProvider);
+            expect(b.container.Thing instanceof ThingProvider).toBe(true);
```

Comment:
@cjsaylor You should maybe access the `Thing` service twice, and then check that `i` is equal to `1` before calling `resetProviders`, as a way of sorta "proving" that it is cached before resetting it.